compiler/rustc_interface/src/passes.rs

```diff
@@ -805,7 +805,8 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
     sess.time("misc_checking_2", || {
         parallel!(
             {
-                tcx.ensure().effective_visibilities(());
+                // Prefetch this as it is used later by lint checking and privacy checking.
+                let _ = tcx.effective_visibilities(());
```

Comment:
Why isn't the `ensure` enough? `effective_visibilities` is eval_always.

Reply:
I've changed all the prefetching to use `ensure_with_value`. It's nicer than having to rely on queries being `eval_always` which might possibly change.